tests/Feature/DashboardTest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace AndreasElia\Analytics\Tests\Unit;
+
+use Illuminate\Http\Request;
+use AndreasElia\Analytics\Tests\TestCase;
+use AndreasElia\Analytics\Models\PageView;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use AndreasElia\Analytics\Http\Middleware\Analytics;
+
+class DashboardTest extends TestCase
+{
+    use RefreshDatabase;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        PageView::create([
+            'session' => '123',
+            'uri' => '/test',
+            'source' => 'example.com',
+            'country' => 'us',
+            'browser' => 'chrome',
+            'device' => 'desktop',
+            'created_at' => now(),
+        ]);
+
+        PageView::create([
+            'session' => '123',
+            'uri' => '/test2',
+            'source' => 'example.com',
+            'country' => 'us',
+            'browser' => 'chrome',
+            'device' => 'desktop',
+            'created_at' => now(),
+        ]);
+
+        PageView::insert([
+            'session' => '5555',
+            'uri' => '/test2',
+            'source' => 'example.com',
+            'country' => 'us',
+            'browser' => 'chrome',
```

Comment:
Is there a reason for the mixture of create/insert use here? Quite a lot of the data is duplicate too, so I wonder if we could define a simple data array and reuse that data and append the data that changes between each insert?

Reply:
The reason for insert is to avoid laravel overwriting the created at date. But they could be identical if you prefer

Reply:
Maybe we could also do something like this:

```php
$testData = [
    'session' => '123',
    'uri' => '/test',
    'source' => 'example.com',
    'country' => 'us',
    'browser' => 'chrome',
    'device' => 'desktop',
    'created_at' => now(),
];

for ($i = 0; $i < 20; $i++) {
    PageView::create($testData);
}
```

And for random intervals we could update the relevant property in the loop?

Just a though since most of the inserts have repeated data.

Reply:
Yeah I will try and clean it up. I used static data to make sure the numbers were exact and predictable (session and created_at)

Reply:
I am just looking at it now, and I am thinking I should just add a model factory. That would clean it up a lot. Is that fine with you?

Reply:
With a factory I have reduced it to this
```
        PageViewFactory::new()->count(2)->create([
            'session' => 'abc123'
        ]);

        $this->travelTo(now()->subDay());
        PageViewFactory::new()->create();

        $this->travelTo(now()->subDays(5));
        PageViewFactory::new()->count(2)
            ->sequence(
                ['uri' => '/test1'],
                ['uri' => '/test2']
            )
            ->create(['session' => 'foo']);

        $this->travelTo(now()->subWeeks(3));
        PageViewFactory::new()->create();

        $this->travelBack();

Reply:
Awesome, nice work @sinnbeck 👍

Reply:
Absolutely. I have never used styleci but I thought that was an external service. Let me just read up on it so I can figure out how to apply it :)

Reply:
No problem. If you scroll down to the bottom of the PR you should see "continuous-integration/styleci/pr", if you click on the "Details" link next to that, it will show you the styling fixes required.

As a side note, it's free for open source projects which makes it very handy for maintaining a consistent code style between repositories 😁

Reply:
Managed to merge it in. Hope thats fine :)

Reply:
Looks great, thank you!

Reply:
I'll get this merged in later on today 😁